docs/book/content/tracing/index.md

```diff
@@ -0,0 +1,296 @@
+
+Manually setting `#[graphql(tracing(ignore))]` to avoid tracing of all, let's say for
+example synchronous field resolvers is rather inefficient when you have GraphQL
+object with too much fields. In this case you can use `tracing` attribute on
+top-level to skip tracing of specific field group or not to trace at all.
+`tracing` attribute can be used with one of the following arguments:
+`sync`, `async`, `only` or `skip_all`.
+ - Use `sync` to trace only synchronous resolvers (struct fields and `fn`s).
+ - Use `async` to trace only asynchronous resolvers (`async fn`s) and
+subscriptions.
+ - Use `only` to trace only fields marked with `#[graphql(tracing(only))]`
+ - Use `skip_all` to skip tracing of all fields.
+
+### Example
+
+```rust
+# extern crate juniper;
+# use juniper::graphql_object;
+# fn main() {}
+
+struct MagicOfTracing;
+
+#[graphql_object]
+#[tracing(async)]
+impl MagicOfTracing {
+    // Won't produce span because it's sync resolver
+    fn my_sync_fn(&self) -> String {
+        "Woah sync resolver!!".to_owned()
+    }
+
+    // Will produce span `MagicOfTracing.myAsyncFn`.
+    async fn my_async_fn(&self) -> String {
+        "Woah async resolver with traces!!".to_owned()
+    }
+
+    // Won't produce span because even though this is an async resolver
+    // it's also marked with `#[graphql(tracing(ignore))]`.
+    #[graphql(tracing(ignore))]
+    async fn non_traced_async_fn(&self) -> String {
+        "Leave no traces".to_owned()
+    }
+}
+```
+
+**Note:** using of `tracing(sync)` with derived struct is no-op because all
+resolvers within derived GraphQL object is considered to be synchronous, also
+because of this `tracing(async)` will result in no traces.
+
+In addition you can use `#[graphql(tracing(ignore))]` with `sync` and `async`
+variants to exclude field from tracing even if it belongs to traced group.
+
+**Be careful when skipping trace as it can lead to bad structured span trees,
+disabling of tracing on one level won't disable tracing in it's child methods.
+As a rule of thumb you should trace all field resolvers which may produce child
+spans.**
+
+If resolving of certain field requires additional arguments (when used `fn`s or
+`async fn`s) they also will be included in resulted trace (except `self` and
+`Context`).
+
+```rust
+# extern crate juniper;
+# use juniper::{graphql_object, GraphQLObject};
+#
+# fn main() {}
+#
+# type Filter = i32;
+#
+# #[derive(GraphQLObject)]
+# struct Product {
+#     id: i32   
+# }
+#
+# struct Catalog;
+#[graphql_object]
+impl Catalog {
+    async fn products(filter: Filter, count: i32) -> Vec<Product> {
+        // Some query
+# unimplemented!()
+    }
+}
+```
+
+In example above both `filter` and `count` of `products` field will be recorded
+in produced [`Span`]. All fields will be recorded using their `fmt::Debug`
+implementation, if your field doesn't implement `fmt::Debug` you'll get compile
+time error. In this case ypu should either implement `fmt::Debug` or skip it
```

Comment:
```suggestion
time error. In this case you should either implement `fmt::Debug` or skip it
```

Reply:
@mathroc Thanks for Your help, I'll fix this with next commit.